umbraco/Umbraco-CMS
# Add functionality for resetting to the initial state.

### Prerequisites


If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/10778
### Description
This PR adds functionality for resetting the VM to the initial state, instead of keeping the modified VM in use.

Steps to validate functionality: 

1. Open Permissions dialog. 
2. Open a user group and click some permissions
3. Click cancel
4. Reopen user group and verify values that were cancelled are not still set

- [x] Tested adding and resetting permissions to node with no preset permissions
- [x] Tested adding and resetting permissions to node with preset permissions
- [x] Tested adding and resetting permissions to node after saving permissions and keeping dialog open.

![HHs6EVYZOc](https://user-images.githubusercontent.com/6340507/140642870-1d91b69f-4e68-4789-bce4-b622fb4f206a.gif)
